### PR TITLE
Bunyanlog implementation (same as spark-server)

### PR DESCRIPTION
**Pull request by haeferer**
I also rebuild all log entries, as good as possible.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,6 @@
     ]
   },
   "pre-commit": [
-    "lint-staged",
```
Comment (AntonPuko) [inline package.json]: ???

File: src/clients/Device.js [hunk 2]
```diff
@@ -319,14 +320,14 @@ class Device extends EventEmitter {
...
-      logger.log('On device protocol initialization complete:\r\n', {
+      logger.info({
         cache_key: this._connectionKey,
         deviceID: this.getDeviceID(),
         firmwareVersion: this._attributes.productFirmwareVersion,
         ip: this.getRemoteIPAddress(),
         platformID: this._attributes.platformId,
         productID: this._attributes.particleProductId,
-      });
+      },'On device protocol initialization complete:\r\n');
```
Comment (AntonPuko) [inline src/clients/Device.js]: if bunyan returns json, we don't need `\r\n` here anymore

File: src/clients/Device.js [hunk 3]
```diff
@@ -811,15 +813,13 @@ class Device extends EventEmitter {
...
     if (this._owningFlasher === flasher) {
       this._owningFlasher = null;
     } else if (this._owningFlasher) {
       logger.error(
-        'cannot releaseOwnership, ',
-        flasher,
-        " isn't the current owner ",
-        { deviceID: this.getDeviceID() },
+        { deviceID: this.getDeviceID(), flasher },
+        'cannot releaseOwnership, isn\'t  current owner'
```
Comment (AntonPuko) [inline src/clients/Device.js]: there should be trailing comma here, so you defenitly didn't run percommit hooks for the pr.

File: src/lib/logger.js [hunk 4]
```diff
@@ -1,73 +1,42 @@
-/*
-*   Copyright (C) 2013-2014 Spark Labs, Inc. All rights reserved. -  https://www.spark.io/
```
Comment (AntonPuko) [inline src/lib/logger.js:2]: i think we can remove this copyright from the file.

File: package.json (hunk 1 above)
Comment (haeferer) [reply package.json]: Corrected, my failure (i had problems with pre-commit lint  #121)

File: src/clients/Device.js (hunk 2 above)
Comment (haeferer) [reply src/clients/Device.js]: Corrected. i've also changed a few others

File: src/clients/Device.js (hunk 3 above)
Comment (haeferer) [reply src/clients/Device.js]: What do you MEAN (sorry, but there seems to be a problem or a missing configuration)

Let us find the Problem and not talking about "you didn't do something" or "did you run tests".
I will do my best to deliver a good result, and will try to fix all problems!

### pre-commit (from my site, not only mentioned in package.json!)
 
* npm run lint-staged
* npm run lint
* npm run test
* flow check

Everything was DONE with THIS Result (right now!)

```
$ npm run lint

> spark-protocol@0.1.6 lint D:\keastick\RemoteRep\spark-protocol
> eslint --fix --max-warnings 0 -- .


$ npm run lint-staged

> spark-protocol@0.1.6 lint-staged D:\keastick\RemoteRep\spark-protocol
> lint-staged

 ↓ Running tasks for src/**/*.js [skipped]
   → No staged files match src/**/*.js
 ↓ Running tasks for test/**/*.js [skipped]
   → No staged files match test/**/*.js
 
npm run test

> spark-protocol@0.1.6 test D:\keastick\RemoteRep\spark-protocol
> ava --no-cache


  √ should subscribe to event
  √ should listen for public event from another owner device
  √ should filter private event
  √ should filter internal event
  √ should filter event by connectionID
  √ should filter event by deviceID
  √ should filter broadcasted events
  √ should listen for mydevices events only
  √ should unsubscribe all subscriptions by subsriberID
  √ should publish and listen for response

  10 tests passed [09:34:15]

$ flow check

src/lib/Handshake.js:325
325:         deviceProvidedPem,
             ^^^^^^^^^^^^^^^^^ null. This type is incompatible with the expected param type of
 91:     publicKeyPem: string,
                       ^^^^^^ string. See: src/lib/CryptoManager.js:91

src/lib/Handshake.js:325
325:         deviceProvidedPem,
             ^^^^^^^^^^^^^^^^^ undefined. This type is incompatible with the expected param type of
 91:     publicKeyPem: string,
                       ^^^^^^ string. See: src/lib/CryptoManager.js:91

src/scripts/update-firmware-binaries.js:82
 82:     file.on('finish', (): void => file.close((): void => resolve(filename)));
                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unused function argument
  728:     close(): void
           ^^^^^^^^^^^^^ function type expects no arguments. See lib: C:\Users\ahr\AppData\Local\Temp\flow\flowlib_c41dfb6\node.js:728


Found 3 errors
```

Comment (haeferer) [reply src/clients/Device.js]: I added the , and will try to figure out the error

File: src/lib/logger.js (hunk 4 above)
Comment (haeferer) [reply src/lib/logger.js:2]: Done

File: src/clients/Device.js (hunk 3 above)
Comment (haeferer) [reply src/clients/Device.js]: The reason is simple. These are parameters.

```
function (
   a,
   b,
)
```
a trailing comma after 'b' is allowed but NOT forced by the current coding style.

So  configuration of eslint needs to be changed to force usage of trailing commas on function parameters

I will make a PR for this

**Comment by haeferer** [comment]
Done, please review

File: src/clients/Device.js (hunk 3 above)
Comment (haeferer) [reply src/clients/Device.js]: I've tried a little bit, but is not easy. So please decide if trailing commas on parameters should be used. 
Here is a warning from ESLINT

* functions is set to "ignore" by default for consistency with the string option.
* functions should only be enabled when linting ECMAScript 2017 or higher.

http://eslint.org/docs/rules/comma-dangle

**Review by jlkalberer — APPROVED** [review]
Looks good to me. Look over it @AntonPuko and then we merge.

File: src/clients/Device.js (hunk 3 above)
Comment (AntonPuko) [reply src/clients/Device.js]: Hey, sorry if I was rude, but when some lines before I see commented lint-staged task, I can't react another way :). Don't go crazy about those commas, we'll figure it out. I thought that `prettier --trailing-comma all` should add all the commas. Anyways, big thanks for your contributions.

Comment (haeferer) [reply src/clients/Device.js]: Ok, let's do our best :)

**Pull request merged**